File: Dddml.Wms.JavaCommon/src/org/dddml/wms/domain/OrganizationStateEventIdFlattenedDto.java
package org.dddml.wms.domain;

import java.util.HashMap;
import java.util.Map;

public class OrganizationStateEventIdFlattenedDto
{


    public static final String[] propertyNames = new String[]{
            "organizationId",
            "version",
        };

    public static final String[] propertyTypes = new String[]{
            "String",
            "Long",
        };

    public static final Map<String, String> propertyTypeMap;

    static {
        propertyTypeMap = new HashMap<String, String>();
        for (int i = 0; i < propertyNames.length; i++ ) {
            propertyTypeMap.put(propertyNames[i], propertyTypes[i]);
        }
    }
	
    private OrganizationStateEventId value;

    public OrganizationStateEventIdFlattenedDto()
    {
        this(new OrganizationStateEventId());
    }

    public OrganizationStateEventIdFlattenedDto(OrganizationStateEventId value)
    {
        this.value = value;
    }

    public OrganizationStateEventId toOrganizationStateEventId()
    {
        return this.value;
    }


    public String getOrganizationId()
    {
        return this.value.getOrganizationId();
    }

    public void setOrganizationId(String organizationId)
    {
        this.value.setOrganizationId(organizationId);
    }

    public Long getVersion()
    {
        return this.value.getVersion();
    }

    public void setVersion(Long version)
    {
        this.value.setVersion(version);
    }


    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        OrganizationStateEventIdFlattenedDto other = (OrganizationStateEventIdFlattenedDto)obj;
        return value.equals(other.value);
    }

    @Override
    public int hashCode()
    {
        return value.hashCode();
    }

}

File: Dddml.Wms.HttpServices.PhpClient/Dddml/Wms/Domain/OrganizationStateEventIdFlattenedDto.php
<?php

namespace Dddml\Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\Serializer\Type\Long;
use Dddml\FlattenedDtoInterface;

class OrganizationStateEventIdFlattenedDto implements FlattenedDtoInterface
{

    const PROPERTIES = [
            'organizationId' => 'string',
            'version' => 'Long',
        ];

    public static function getPropertyNames()
    {
        return array_keys(static::PROPERTIES);
    }

    public static function getPropertyTypes()
    {
        return array_values(static::PROPERTIES);
    }

	
    /**
     * @var OrganizationStateEventId
     */
    private $value;

    /**
     * @param OrganizationStateEventId $value
     */
    public function __construct(OrganizationStateEventId $value = null)
    {
        if ($value) {
            $this->value = $value;
        } else {
            $this->value = new OrganizationStateEventId();
        }
    }

    /**
     * @return OrganizationStateEventId
     */
    public function toOrganizationStateEventId()
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function getOrganizationId()
    {
        return $this->value->getOrganizationId();
    }

    /**
     * @param string $organizationId
     */
    public function setOrganizationId($organizationId)
    {
        $this->value->setOrganizationId($organizationId);
    }

    /**
     * @return Long
     */
    public function getVersion()
    {
        return $this->value->getVersion();
    }

    /**
     * @param Long $version
     */
    public function setVersion($version)
    {
        $this->value->setVersion($version);
    }

    /**
     * @return string
     */
    public function toString()
    {
        $pValues = [
            $this->getOrganizationId(),
            $this->getVersion(),
        ];
        return implode(',', $pValues);
    }

    /**
     * @param string $data
     *
     * @return OrganizationStateEventIdFlattenedDto
     */
    public function fromString($data)
    {
        $pValues = explode(',', $data);
        $this->setOrganizationId($pValues[0]);
        $this->setVersion($pValues[1]);
        return $this;
    }

}

